example/image-classification/predict-cpp/CMakeLists.txt

```diff
@@ -1,34 +1,28 @@
-if(USE_OPENCV)
-  find_package(OpenCV QUIET COMPONENTS core highgui imgproc imgcodecs)
-  if(NOT OpenCV_FOUND) # if not OpenCV 3.x, then imgcodecs are not found
-    find_package(OpenCV REQUIRED COMPONENTS core highgui imgproc)
-  endif()
-
-  if(NOT MSVC)
-    set(UNITTEST_STATIC_LINK ON)
-  endif()
+# ---[ OpenCV
+if(NOT USE_OPENCV OR NOT OpenCV_FOUND)
+  message(WARNING "\
+OpenCV should be enabled and found to build image classification example, skipping...")
```

Comment:
This is a critical error, right?

Reply:
The example will just not be build and it will not break the whole project configuration.

Reply:
This certainly is chopping out a lot of stuff.  WHat's the net change in static linking logic?

Reply:
It is not about static, it's an early return from this cmake file. The example can not be built if there is no OpenCV and will not be added to the main project. That's it.

Reply:
isn’t the unittest static link flag removed?

Reply:
UNITTEST_STATIC_LINK is descriptive and editable.  Changing the check to MSVC at the actual interpretation point makes it difficult to see what's being done.
Also, I think MSVC has to build static because of dllimport issues

Reply:
I can leave the variable itself but would rename it, because it's an example, not a unit test.